sdks/python/apache_beam/metrics/internal.py

```diff
@@ -0,0 +1,115 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
```

Comment:
"internal.py" is a very vague name.  Because this file contains classes that are used for collecting metrics during pipeline execution, I suggest renaming this class "execution.py".

Additionally, I suggest moving the MetricKey and MetricResult classes from "base.py" to this file (and renaming "base.py" to "metricbase.py" as described in the comments on that file).